### PR TITLE
unroll meta programming in meta

### PR DESCRIPTION
**Pull request by bekzod**
as requested here: https://github.com/emberjs/ember.js/pull/15339#issuecomment-307115452

### CODE REVIEW

File: packages/ember-metal/lib/meta.js [hunk 1]
```diff
@@ -359,28 +350,34 @@ export class Meta {
...
-
-// Implements a member that is a lazily created, non-inheritable
-// POJO.
-function ownMap(name, key, Meta) {
-  Meta.prototype[`writable${name}`] = function() {
-    return this._getOrCreateOwnMap(key);
-  };
-  Meta.prototype[`readable${name}`] = function() { return this[key]; };
-}
+memberNames.forEach(name => members[name](name, Meta));
```
Comment (rwjblue) [inline packages/ember-metal/lib/meta.js]: Does this unroll your previous change?

File: packages/ember-metal/lib/meta.js [hunk 2]
```diff
@@ -386,60 +357,123 @@ export class Meta {
...
-if (EMBER_GLIMMER_DETECT_BACKTRACKING_RERENDER || EMBER_GLIMMER_ALLOW_BACKTRACKING_RERENDER) {
-  Meta.prototype.writableLastRendered = function() { return this._getOrCreateOwnMap('_lastRendered'); };
-  Meta.prototype.readableLastRendered = function() { return this._lastRendered; };
-  if (has('ember-debug')) { //https://github.com/emberjs/ember.js/issues/14732
-    Meta.prototype.writableLastRenderedReferenceMap = function() { return this._getOrCreateOwnMap('_lastRenderedReferenceMap'); };
-    Meta.prototype.readableLastRenderedReferenceMap = function() { return this._lastRenderedReferenceMap; };
-    Meta.prototype.writableLastRenderedTemplateMap = function() { return this._getOrCreateOwnMap('_lastRenderedTemplateMap'); };
-    Meta.prototype.readableLastRenderedTemplateMap = function() { return this._lastRenderedTemplateMap; };
+  deleteFromWatching(subkey) {
+   delete this._getOrCreateOwnMap('_watching')[subkey];
```
Comment (rwjblue) [inline packages/ember-metal/lib/meta.js]: I think the indentation is off a bit here...

File: packages/ember-metal/lib/meta.js [hunk 3]
```diff
@@ -386,60 +357,123 @@ export class Meta {
...
+  }
+
+  forEachMixins(fn) {
+    let pointer = this;
+    let seen;
+    while (pointer !== undefined) {
+      let map = pointer._mixins;
+      if (map !== undefined) {
+        for (let key in map) {
+          seen = seen || Object.create(null);
```
Comment (rwjblue) [inline packages/ember-metal/lib/meta.js]: Should we use `create` here too (I think we do `const { create } = Object;` above)?

File: packages/ember-metal/lib/meta.js [hunk 4]
```diff
@@ -412,75 +442,122 @@ function inheritedMap(name, key, Meta) {
...
+  hasInValues(subkey) {
+    return this._findInherited('_values', subkey) !== undefined;
+  }
+
+}
+
+if (EMBER_GLIMMER_DETECT_BACKTRACKING_RERENDER || EMBER_GLIMMER_ALLOW_BACKTRACKING_RERENDER) {
+  Meta.prototype.writableLastRendered = function() { return this._getOrCreateOwnMap('_lastRendered'); };
+  Meta.prototype.readableLastRendered = function() { return this._lastRendered; };
+  if (has('ember-debug')) { //https://github.com/emberjs/ember.js/issues/14732
```
Comment (rwjblue) [inline packages/ember-metal/lib/meta.js]: This can be `if (DEBUG) {`

File: packages/ember-metal/lib/meta.js (hunk 3 above)
Comment (bekzod) [inline packages/ember-metal/lib/meta.js]: couldn't find usage of `const { create } = Object;`

Comment (rwjblue) [inline packages/ember-metal/lib/meta.js]: Hmm, ok, I guess I just thought the `create(this.source)` invocations that are used here were using `Object.create`?

Comment (bekzod) [inline packages/ember-metal/lib/meta.js]: that was passed function param

Comment (rwjblue) [inline packages/ember-metal/lib/meta.js]: aha! sorry for the confusion!

Comment (bekzod) [inline packages/ember-metal/lib/meta.js]: np 👍

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**